### DISCUSSION TITLE
Visual Studio - Publish AWS Serverless Application - Forced S3 Encryption policy

### DISCUSSION THREAD

**Discussion by FritsDonk** [aws/aws-lambda-dotnet#1755] [Q&A] [answered]
Hi all,

I am fairly new to AWS and have run into a problem. I am trying to publish from Visual Studio a AWS Serverless Application.
The publish process asks for an S3 Bucket to upload the ZIP to, but this fails with Access Denied.

The reason for this is because we have a SCP policy that Denies S3:PutObject requests if they do not use encryption (s3:x-amz-server-side-encryption). I cannot find any information or assistance in specifying this in the publish settings, or in the Cloudformation serverless.template file.

I saw another Q&A discussion which mentioned it uses the S3 Backup temporarily to upload the ZIP file for the CreateFunction bits on AWS. But it mentioned that it can use MemoryStream BLOB if not bucket is provided, but I also cannot find this.

Please help.

**Comment by ashovlin** [comment, marked as answer]
We have a related feature request for this on the CLI deployment tool (which the Visual Studio Toolkit uses under the hood): https://github.com/aws/aws-extensions-for-dotnet-cli/issues/94

**Reply by FritsDonk** [reply]
Thanks for letting me know.
